lib/daru/view/adapters/highcharts/display.rb

```diff
@@ -61,6 +64,47 @@ def to_html_iruby(placeholder=random_canvas_id)
       high_chart_iruby(extract_chart_class, placeholder, self)
     end
 
+    # Loads the dependent mapdata and dependent modules of the chart
+    #
+    # @param [String] to determine whether to load modules in IRuby or web
+    #   frameworks
+    # @return [void, String] loads the initial script of the modules for IRuby
+    #   notebook and returns initial script of the modules for web frameworks
+    def load_dependencies(type)
+      dep_js = extract_dependencies
+      if type == 'iruby'
+        LazyHighCharts.init_iruby(dep_js) unless dep_js.nil?
+      elsif type == 'web_frameworks'
+        dep_js.nil? ? '' : LazyHighCharts.init_script(dep_js)
+      end
+    end
+
+    # Extracts the required dependencies for the chart. User does not need
+    #   to provide any mapdata requirement explicity in the `options`.
+    #   MapData will be extracted using `options[:chart][:map]` already
+    #   provided by the user. In `modules` user needs to provide the required
+    #   modules (like tilemap in highcharts) in the form of Array. Once the
+    #   dependency is loaded on a page, there is no need to provide it again in
+    #   the `modules` option.
+    #
+    # @return [Array] the required dependencies (mapdata or modules)
+    #   to load the chart
+    # rubocop:disable Metrics/AbcSize
+    def extract_dependencies
+      dep_js = []
+      # Mapdata dependencies
+      if !options[:chart_class].nil? && options[:chart_class].capitalize == 'Map'
+        dep_js.push('mapdata/' + options[:chart][:map].to_s + '.js') if
+        options[:chart] && options[:chart][:map]
+      end
+      # Dependencies provided in modules option (of highcharts mainly
+      #   like tilemap) by the user
+      dep_js |= options.delete(:modules).map! { |js| "#{js}.js" } unless
```

Comment:
`map` option is not created by us and will be provided in highcharts' options everytime there is a need for map data(like europe.js). This would be the exact address of the dependency required.
For example, a user using dependency `https://code.highcharts.com/mapdata/countries/in/in-all.js` will provide `countries/in/in-all` as a value for options[:chart][:map]. 
Only the options `chart_class` and `modules` are created by us.
Yes, we can have a third parameter for these but IMO that would be only more complex for the user plus we then have to generalize it for other adapters as well which I think is not a good option.

Reply:
`modules` option is created just to load some additional dependencies used in a specific chart by the user.